src/lib.rs

```diff
@@ -486,6 +517,62 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    #[cfg(unix)]
+    // TODO: Do Android and MacOS implement this feature of utimensat?
```

Comment:
Android: Yes.
- https://github.com/aosp-mirror/platform_bionic/blob/d06e2e7b29ec785b9c9143b98c04dd75286fc5d5/libc/include/sys/stat.h#L178-L192

Darwin: Not immediately clear at a glance.
- https://github.com/apple-oss-distributions/xnu/blob/94d3b452840153a99b38a3a9659680b2a006908e/bsd/sys/stat.h#L582-L583
- https://github.com/apple-oss-distributions/xnu/blob/94d3b452840153a99b38a3a9659680b2a006908e/tests/utimensat.c
- https://github.com/apple-oss-distributions/xnu/blob/94d3b452840153a99b38a3a9659680b2a006908e/tests/netbsd_utimensat.c

Reply:
Then I guess it's safest to implement it just the way that this PR already does: Defer to the non-NULL variant, and improve it in a later PR if/when more is known about what Darwin accepts.